### PR TITLE
Docs: Concepts section

### PR DESCRIPTION
**Pull request by AdrianoKF**
Adds a "Concepts" section to the documentation:

- High-level architecture
- Job lifecycle overview
- Job identifiers

Also, the landing page experience was slightly improved by introducing a card grid.

### CODE REVIEW

File: client/docs/concepts/identifiers.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+Kueue handles the `Workload` and updates its status to reflect the current state of the workload.
+
+![Kueue workload components](https://kueue.sigs.k8s.io/images/queueing-components.svg)
+
+## Identifying workloads
+
+Every Workload (as managed by Kueue) carries by an automatically generated unique identifier (UID) as well as a human-readable name and namespace.
+Both these could serve as a unique identifier for a Workload. However, a name/namespace combination is not guaranteed to be unique over time (for example when deleting and recreating), whereas UIDs are.
+This which make UIDs a slightly better choice for identifying a given Workload resource.
```
Comment (nicholasjng) [inline client/docs/concepts/identifiers.md:23]:
```suggestion
This makes UIDs a slightly better choice for identifying a given Workload resource.
```

File: client/docs/concepts/identifiers.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+The concrete workload resource has the same identifiers, a UID and name/namespace combination.
+
+A given concrete workload resource references its associated abstract workload in a 1:1 fashion (through its `metadata.ownerReferences` field).
```
Comment (nicholasjng) [inline client/docs/concepts/identifiers.md:27]: This is potentially confusing - I assume "workload resource = `Workload`" (Kueue) holds here?

File: client/docs/concepts/identifiers.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+```mermaid
+graph LR
+subgraph "Namespace example"
+direction LR
+  A["`Workload **job-example** <pre>uid-1</pre>`"] --> B["`Job **example** <pre>uid-2</pre>`"] --> C[Pod]
+end
+```
+
+If we want to want to query the logs of the workload, we can do so by calling `jobq logs` with the UID of the concrete workload resource:
```
Comment (nicholasjng) [inline client/docs/concepts/identifiers.md:47]:
```suggestion
If we want to query the logs of the workload, we can do so by calling `jobq logs` with the UID of the concrete workload resource:
```

File: client/docs/concepts/lifecycle.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+The remainder of this document uses the terms _job_ and _workload_ interchangeably.
+
+A workload roughly goes through three phases after its submission: _queuing and scheduling_, _execution_, and _completion_.
+
+### Queueing and scheduling
+
+After its submission, a workload is in the `Submitted` state, where it competes with other workloads for available resource quotas.
+Once it is admitted to a cluster queue, it enters the `Pending` state, where Kueue will reserve a quota for it.
+Alternatively, if the selected local or cluster queue for the workload are stopped, the workload will enter the `Inadmissible` state until this condition is resolved.
```
Comment (nicholasjng) [inline client/docs/concepts/lifecycle.md:14]: (or do not exist)

File: client/docs/concepts/lifecycle.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+### Execution
+
+After all admission checks for the workload have passed, it enters the `Admitted` state, it is now eligible for execution by the cluster.
+
+### Completion
+
+When the workload terminates successfully, it enters the terminal `Succeeded` state.
+If any unrecoverable error occurs during execution, the workload enters the terminal `Failed` state.
```
Comment (nicholasjng) [inline client/docs/concepts/lifecycle.md:23]: Maybe it's worth highlighting that depending on the completion/backoff policy, "my pod has failed" does not equal "my job has failed". (Thinking back to our list pair programming session)

File: client/docs/concepts/identifiers.md (hunk 2 above)
Comment (AdrianoKF) [reply client/docs/concepts/identifiers.md]: Improved the wording in 431d3c2a0dbe8889f31d1a291872197b422cb89a.

File: client/docs/concepts/lifecycle.md (hunk 5 above)
Comment (AdrianoKF) [reply client/docs/concepts/lifecycle.md]: Improved wording in 4d1579d892f8d9e8027539d3021d056f7c461bf0

**Pull request merged**